src/software/sensor_fusion/sensor_fusion.cpp

```diff
@@ -87,10 +106,14 @@ void SensorFusion::updateWorld(const SSLProto::Referee &packet)
     if (sensor_fusion_config->FriendlyColorYellow()->value())
     {
         game_state.updateRefereeCommand(createRefereeCommand(packet, TeamColour::YELLOW));
+        friendly_goalie_id = packet.yellow().goalkeeper();
+        enemy_goalie_id    = packet.blue().goalkeeper();
     }
     else
     {
         game_state.updateRefereeCommand(createRefereeCommand(packet, TeamColour::BLUE));
+        friendly_goalie_id = packet.blue().goalkeeper();
+        enemy_goalie_id    = packet.yellow().goalkeeper();
```

Comment:
Do we need to store the goalie ID here? Can we do:

~~~c++
  friendly_team.assignGoalie(packet.yellow().goalkeeper());
  enemy_team.assignGoalie(packet.blue().goalkeeper());
~~~

Then we can remove the static variables, the goalie id overriding happens in `processSensorProto` after `updateWorld` is called so it should be good

Reply:
I left the variables there so I could initialise a default value of 1, in the (possibly rare) case that we don't receive a referee message and the goalie isn't overridden. Not sure if there is a better way of doing this, or if this is necessary?

Reply:
They can be initialized in the SensorFusion constructor to 0